### PR TITLE
Use pip-tools to pin transitive dependencies and dependabot for version updates

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

We had a breakage today due to a release of a transitive dependency. This PR ports our requirements to be generated with pip-tools to pin *all* versions. Also added dependabot to notify us when versions need updating.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Ran locally
* Ran tests
* Ran azd up

### CODE REVIEW

File: app/backend/requirements.in [hunk 1]
```diff
@@ -0,0 +1,15 @@
+azure-identity
+quart
+langchain
+openai[datalib]
+tiktoken
+azure-search-documents==11.4.0b6
```
Comment (mattmsft) [inline app/backend/requirements.in:6]: 11.5.0-beta.4 is the latest
https://www.nuget.org/packages/Azure.Search.Documents/11.5.0-beta.4

Comment (mattmsft) [reply app/backend/requirements.in:6]: but if this causes some breaks we shouldn't pull it in

**Review by mattmsft — APPROVED** [review]

File: app/backend/requirements.in (hunk 1 above)
Comment (pamelafox) [reply app/backend/requirements.in:6]: For Python, we seem to have the latest? https://pypi.org/project/azure-search-documents/#history

Comment (pamelafox) [reply app/backend/requirements.in:6]: (This is the version we had before, by the way. I had to pin it as it's considered a pre-release version, but we need it for HnswParameters)

Comment (mattmsft) [reply app/backend/requirements.in:6]: ah, looking at the wrong feed. Yes, you are right!

**Pull request merged**